### ISSUE TITLE
Subdomains and the route decorator

### ISSUE THREAD

**Issue by DeaconDesperado** [apiguy/flask-classy#6]
This may or may not be intended functionality, though I assume it is worth it to check.

Currently, I have a FlaskView called PlaceView to provide HTTP access to a lot of my Place model methods.  I need to register this view at both the main domain (foo.com) at route_base '/data' as well as the data.foo.com subdomain.  I do this using the following utility function:

```
def attach_classy(view,app=app):
    view_base = '/data%s' % view.route_base
    print 'Attaching %s at %s' % (view,view_base)
    view.register(app,route_base=view_base)
    view.register(app,subdomain='data')
```

Where the parameter `view` is the FlaskView passed in (I use this in a loop to register all my views as such)

The problem is that any view methods routed using the `@route` decorator will not register at the subdomain aliased route.  Thus a method like this:

```
@route('/<id>/posts',methods=['GET'])
def getraves(self,id):
    try:
        place_id = ObjectId(id)
        raves = Posts.collection.find({'rave_type':1,'entity_id':id})
        return Response([json.dumps([r for r in raves])],mimetype='application/json')
    except:
        return Response([json.dumps({'flag' :False})], mimetype='application/json')
```

Will generate these rules (used regex to find them in Map.iter_rules() )

```
/data/place/<id>/posts
/data/place/<id>/posts
/data/place/<id>/posts
/data/place/<id>/posts
/place/<id>/posts
/place/<id>/posts
```

But no such rule:

```
data|/place/<id>/posts
```

When registered using the first function above.

So, the `subdomain` kwarg used in register doesn't seem to matter in the case of `@route` decorated methods.  Is this intended?  Should the proper remedy be two lines of `@route` decoration, one with subdomain and one without?  Perhaps a list of subdomains, one of which could be ''?

**Comment by apiguy** [comment]
Definitely not the intended behavior. The subdomain should be picked up even when using the `@route` decorator. I'll look into this now.

**Comment by apiguy** [comment]
The fix is in 0.5.1 for this issue Mark. Thanks for the excellent write-up on the problem.
